### PR TITLE
`ragenix --generate`

### PR DESCRIPTION
**Pull request by blaggacao**
- Tooling: change from naersk to nixpkgs+fenix
- Tooling: avoid double building between codeStyleConformanceCheck & cargoCheckHook
- Feature: add first implementation of secrets generate script

### CODE REVIEW

**Comment by blaggacao** [comment]
This PR is an initial iteration for a `ragenix --generate` implementation. As a first iteration, this currently depends on that binaries used are made available within the environment, e.g. via a devshell.

/cc @ryantm fyi

File: src/cli.rs [hunk 1]
```diff
@@ -63,9 +64,18 @@ fn build() -> App<'static> {
                 .short('s')
                 .takes_value(false),
         )
+        .arg(
+            Arg::new("generate")
+                .about("(re-)generates the age-encrypted FILE using the specified generate script")
```
Comment (veehaitch) [inline src/cli.rs:69]:
```suggestion
                .help("(re-)generates the age-encrypted FILE using the specified generate script")
```

**Comment by blaggacao** [comment]
I realize this approach is a bit incomplete, I haven't made up my mind, but especially cryptographic triplets (cert, key, chain) are very unhandy to manipulate atomically like this.

I'm currently experimenting with spire/spiffe for distribution of cryptographic identities.

Maybe a valid conlusion is to recommend _not_ using ragenix for distributing cryptografic triplets and rather scope to "traditional" userpass credential stuff.

In the end, however, cryptografic identities are the holy grale and replace any need of userpass (etc.) stuff.

Looking at it in that light any agenix-based secrets distribution would have to be considered "legacy" compared to spire/spiffe (or similar).

Not sure, just putting down fuzzy notes.

**Comment by blaggacao** [comment]
@veehaitch I decided that this is ready for review. The problem of cryptographic triplets shouldn't be a problem that ragenix tries to solve within the scope of the generate command.

**Comment by veehaitch** [comment]
This PR has been stale for some time. Let me know if you want to continue working on this feature.

**Pull request closed without merging**